frontend/public/components/utils/file-input.tsx

```diff
@@ -1,50 +1,153 @@
 import * as React from 'react';
+import * as classNames from 'classnames';
+import { NativeTypes } from 'react-dnd-html5-backend';
+// eslint-disable-next-line no-unused-vars
+import { DropTarget, ConnectDropTarget, DropTargetMonitor } from 'react-dnd';
+import withDragDropContext from './drag-drop-context';
 
 export class FileInput extends React.Component<FileInputProps, FileInputState> {
+  constructor(props) {
+    super(props);
+    this.onDataChange = this.onDataChange.bind(this);
+    this.onFileUpload = this.onFileUpload.bind(this);
+  }
+  onDataChange(event) {
+    this.props.onChange(event.target.value);
+  }
+  readFile(file) {
+    const reader = new FileReader();
+    reader.onload = () => {
+      const input = reader.result;
+      this.props.onChange(input, file.name);
+    };
+    reader.readAsText(file, 'UTF-8');
+  }
+  onFileUpload(event) {
+    this.readFile(event.target.files[0]);
+  }
+  render() {
+    const { connectDropTarget, isOver, canDrop, id } = this.props;
+    const klass = classNames('co-file-dropzone-container', {'co-file-dropzone--drop-over': isOver});
+    return (
+      connectDropTarget(
+        <div className="co-file-dropzone">
+          { canDrop && <div className={klass}><p className="co-file-dropzone__drop-text">Drop file here</p></div> }
+
+          <div className="form-group">
+            <label className="control-label" htmlFor={id}>{this.props.label}</label>
+            <div className="modal-body__field">
+              <div className="input-group">
+                <input type="text"
+                  className="form-control"
+                  value={this.props.inputFileName}
+                  aria-describedby={`${id}-help`}
+                  readOnly
+                  disabled />
+                <span className="input-group-btn">
+                  <span className="btn btn-default co-btn-file">
+                    Browse&hellip;
```

Comment:
Assuming `Browse&hellip;` is ellipsis & not an oops :)

Reply:
`&hellip;` stands for **H**orizontal **Ellipsis**, that's an [HTML entity](https://www.toptal.com/designers/htmlarrows/punctuation/horizontal-ellipsis/). It looks like this: `…`